cardinal/cardinal_test.go

```diff
@@ -40,6 +44,634 @@ type Rawbodytx struct {
 	SignerAddress string `json:"signerAddress"`
 }
 
+func (Health) Name() string { return "health" }
+
+type AddHealthToEntityTx struct {
+	TargetID types.EntityID
+	Amount   int
+}
+
+type AddHealthToEntityResult struct{}
+
+func TestForEachTransaction(t *testing.T) {
+	tf := testutils.NewTestFixture(t, nil)
+	world := tf.World
+	type SomeMsgRequest struct {
+		GenerateError bool
+	}
+	type SomeMsgResponse struct {
+		Successful bool
+	}
+
+	assert.NilError(t, cardinal.RegisterMessage[SomeMsgRequest, SomeMsgResponse](world, "some_msg"))
+
+	err := cardinal.RegisterSystems(world, func(wCtx engine.Context) error {
+		someMsg, err := cardinal.GetMessage[SomeMsgRequest, SomeMsgResponse](wCtx)
+		if err != nil {
+			return err
+		}
+		someMsg.Each(wCtx, func(t message.TxData[SomeMsgRequest]) (result SomeMsgResponse, err error) {
+			if t.Msg.GenerateError {
+				return result, errors.New("some error")
+			}
+			return SomeMsgResponse{
+				Successful: true,
+			}, nil
+		})
+		return nil
+	})
+	assert.NilError(t, err)
+	tf.StartWorld()
+
+	// Add 10 transactions to the tx pool and keep track of the hashes that we just cardinal.Created
+	knownTxHashes := map[types.TxHash]SomeMsgRequest{}
+	for i := 0; i < 10; i++ {
+		someMsg, err := cardinal.GetMessageFromWorld[SomeMsgRequest, SomeMsgResponse](world)
+		assert.NilError(t, err)
+		req := SomeMsgRequest{GenerateError: i%2 == 0}
+		txHash := tf.AddTransaction(someMsg.ID(), req, testutils.UniqueSignature())
+		knownTxHashes[txHash] = req
+	}
+
+	// Perform a engine tick
+	assert.NilError(t, world.Tick(context.Background(), uint64(time.Now().Unix())))
```

Comment:
nit: Ticking should be done with tf.DoTick() (any calls to world.Tick in this test file can be replaced with tf.DoTick())

Reply:
There's a deadlock on `TestAddToPoolDuringTickDoesNotTimeout` I commented it with a TODO and created a ticket. https://linear.app/arguslabs/issue/WORLD-883/implement-primitivestorage-for-typetocomponent

All the other ticks are changed.